ui/static/ui/js/learning_resources.jsx

```diff
@@ -390,6 +415,30 @@ define('learning_resources',
         thiz.setState({loaded: true});
       });
     },
+    componentWillReceiveProps: function(nextProps) {
+      var unSavedChangesMessage = "You have unsaved changes. " +
```

Comment:
Can you use a dialog box for the message as @Ferdi mentioned earlier? I think this will also simplify the code in this class since the confirmation dialog code would be mostly outside this class, you wouldn't need to implement `componentWillReceiveProps` anymore.